pkg/aws/aws.go

```diff
@@ -66,8 +82,10 @@ func GetCredentials(ctx context.Context, config map[string]string) (*credentials
 		if err != nil {
 			return nil, errors.Wrap(err, "Failed to create session to initialize Web Identify credentials")
 		}
-		// If we have credentials to use with a Web Identity provider - use those
-		creds = stscreds.NewWebIdentityCredentials(sess, os.Getenv(roleARNEnvKey), "", os.Getenv(webIdentityTokenFilePathEnvKey))
+		svc := sts.New(sess)
+		p := stscreds.NewWebIdentityRoleProvider(svc, os.Getenv(roleARNEnvKey), "", os.Getenv(webIdentityTokenFilePathEnvKey))
+		p.Duration = assumeRoleDuration
```

Comment:
Should we add a comment here to note why we need to update this separately?

Reply:
Good idea. I've moved the code for getting creds with a duration to a helper function and added comments for that function.